django/db/migrations/operations/models.py

```diff
@@ -802,6 +805,7 @@ def state_forwards(self, app_label, state):
         model_state = state.models[app_label, self.model_name_lower]
         indexes = model_state.options[self.option_name]
         model_state.options[self.option_name] = [idx for idx in indexes if idx.name != self.name]
+        state.reload_model(app_label, self.model_name_lower, delay=True)
```

Comment:
Ok, then perhaps the docstring in `test_state.py` needs some update based on your description? Currently it says, "Tests state construction, rendering and modification by operations."

Also, if the tests remain in `test_operations.py`, I guess it could be nice if `set_up_test_model()` isn't required for those tests since that's adding unnecessary database queries. The `StateTests` use `SimpleTestCase` so they're much faster than the the operations tests which inherit `MigrationTestBase` which add more clean up. Maybe a separate `TestOperationsState` class that inherits `SimpleTestCase` could be useful for `test_operations.py`.

Anyway, if you want to fix this up later, I guess we can do that.